NPE caused internal server error when user fetches an empty property store zk node.

### Describe the bug
When fetching an empty property store zk node through the Helix rest API, user get this response,
"HTTP ERROR 500 Request failed."

If check the log, then we observe this error log,

> Caused by: java.lang.NullPointerException
> at java.lang.String.<init>(String.java:614) ~[?:?]
> at org.apache.helix.rest.server.resources.helix.PropertyStoreAccessor.getPropertyByPath(PropertyStoreAccessor.java:82) ~[helix-rest-1.0.2.200.jar:?]

This is not informative and does not align with what we promised that, "if the node is not in the required Helix ZNRecord format, the API returns content directly."

### To Reproduce
Use "...admin/v2/namespaces/{namespace}/clusters/{cluster}/propertyStore/{path}" to fetch an zk node with no content.

### Expected behavior
Throw a meaningful error message.
Note that returning an empty string is also a possible option but it might be confusing. Return No content error would be clearer.

### Additional context
Add any other context about the problem here.

--- a/helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java
+++ b/helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java
@@ -73,19 +73,24 @@ public class PropertyStoreAccessor extends AbstractHelixResource {
     }
     final String recordPath = PropertyPathBuilder.propertyStore(clusterId) + path;
     BaseDataAccessor<byte[]> propertyStoreDataAccessor = getByteArrayDataAccessor();
-    if (propertyStoreDataAccessor.exists(recordPath, AccessOption.PERSISTENT)) {
-      byte[] bytes = propertyStoreDataAccessor.get(recordPath, null, AccessOption.PERSISTENT);
-      ZNRecord znRecord = (ZNRecord) ZN_RECORD_SERIALIZER.deserialize(bytes);
-      // The ZNRecordSerializer returns null when exception occurs in deserialization method
-      if (znRecord == null) {
-        ObjectNode jsonNode = OBJECT_MAPPER.createObjectNode();
-        jsonNode.put(CONTENT_KEY, new String(bytes));
-        return JSONRepresentation(jsonNode);
-      }
-      return JSONRepresentation(znRecord);
-    } else {
+    if (!propertyStoreDataAccessor.exists(recordPath, AccessOption.PERSISTENT)) {
       throw new WebApplicationException(Response.status(Response.Status.NOT_FOUND)
-          .entity(String.format("The property store path %s doesn't exist", recordPath)).build());
+          .entity(String.format("The property store path %s doesn't exist", recordPath))
+          .build());
+    }
+    byte[] bytes = propertyStoreDataAccessor.get(recordPath, null, AccessOption.PERSISTENT);
+    if (bytes == null) {
+      throw new WebApplicationException(Response.status(Response.Status.NO_CONTENT).build());
+    }
+    ZNRecord znRecord = (ZNRecord) ZN_RECORD_SERIALIZER.deserialize(bytes);
+    // The ZNRecordSerializer returns null when exception occurs in deserialization method
+    if (znRecord == null) {
+      // If the zk node cannot be deserialized, return the content directly.
+      ObjectNode jsonNode = OBJECT_MAPPER.createObjectNode();
+      jsonNode.put(CONTENT_KEY, new String(bytes));
+      return JSONRepresentation(jsonNode);
+    } else {
+      return JSONRepresentation(znRecord);
     }
   }
 

--- a/helix-rest/src/test/java/org/apache/helix/rest/server/TestPropertyStoreAccessor.java
+++ b/helix-rest/src/test/java/org/apache/helix/rest/server/TestPropertyStoreAccessor.java
@@ -49,6 +49,8 @@ public class TestPropertyStoreAccessor extends AbstractTestClass {
   private static final ZNRecord TEST_ZNRECORD = new ZNRecord("TestContent");
   private static final String CUSTOM_PATH =
       PropertyPathBuilder.propertyStore(TEST_CLUSTER) + "/NonZnRecord";
+  private static final String EMPTY_PATH =
+      PropertyPathBuilder.propertyStore(TEST_CLUSTER) + "/EmptyNode";
   private static final String TEST_CONTENT = "TestContent";
   private static final String CONTENT_KEY = "content";
 
@@ -71,6 +73,7 @@ public class TestPropertyStoreAccessor extends AbstractTestClass {
     Assert
         .assertTrue(_customDataAccessor.create(CUSTOM_PATH, TEST_CONTENT, AccessOption.PERSISTENT));
     Assert.assertTrue(_baseAccessor.create(ZNRECORD_PATH, TEST_ZNRECORD, AccessOption.PERSISTENT));
+    Assert.assertTrue(_baseAccessor.create(EMPTY_PATH, null, AccessOption.EPHEMERAL));
   }
 
   @AfterClass
@@ -78,6 +81,13 @@ public class TestPropertyStoreAccessor extends AbstractTestClass {
     if (_customDataAccessor != null) {
       _customDataAccessor.close();
     }
+  }
+
+  @Test
+  public void testGetPropertyStoreWithEmptyContent() {
+    String data = new JerseyUriRequestBuilder("clusters/{}/propertyStore/EmptyNode").format(TEST_CLUSTER)
+            .expectedReturnStatusCode(Response.Status.NO_CONTENT.getStatusCode()).get(this);
+    Assert.assertTrue(data.isEmpty());
   }
 
   @Test
@@ -91,9 +101,9 @@ public class TestPropertyStoreAccessor extends AbstractTestClass {
 
   @Test
   public void testGetPropertyStoreWithTestStringData() throws IOException {
-    String actual =
-        new JerseyUriRequestBuilder("clusters/{}/propertyStore/NonZnRecord").format(TEST_CLUSTER)
-            .isBodyReturnExpected(true).get(this);
+    String actual = new JerseyUriRequestBuilder("clusters/{}/propertyStore/NonZnRecord").format(TEST_CLUSTER)
+        .isBodyReturnExpected(true)
+        .get(this);
     JsonNode jsonNode = OBJECT_MAPPER.readTree(actual);
     String payLoad = jsonNode.get(CONTENT_KEY).textValue();
 

--- a/helix-rest/src/test/java/org/apache/helix/rest/server/util/JerseyUriRequestBuilder.java
+++ b/helix-rest/src/test/java/org/apache/helix/rest/server/util/JerseyUriRequestBuilder.java
@@ -94,13 +94,15 @@ public class JerseyUriRequestBuilder {
 
     Assert.assertEquals(response.getStatus(), _expectedStatusCode);
 
-    // NOT_FOUND will throw text based html
-    if (_expectedStatusCode != Response.Status.NOT_FOUND.getStatusCode()
-        && _expectedStatusCode != Response.Status.BAD_REQUEST.getStatusCode()) {
-      Assert.assertEquals(response.getMediaType().getType(), "application");
-    } else {
-      Assert.assertEquals(response.getMediaType().getType(), "text");
-    }
+    if (_expectedStatusCode != Response.Status.NO_CONTENT.getStatusCode()) {
+      // NOT_FOUND will throw text based html
+      if (_expectedStatusCode != Response.Status.NOT_FOUND.getStatusCode()
+          && _expectedStatusCode != Response.Status.BAD_REQUEST.getStatusCode()) {
+        Assert.assertEquals(response.getMediaType().getType(), "application");
+      } else {
+        Assert.assertEquals(response.getMediaType().getType(), "text");
+      }
+    } // else, NO_CONTENT should not return any content, so no need to check the type.
 
     String body = response.readEntity(String.class);
     if (_isBodyReturnExpected) {

